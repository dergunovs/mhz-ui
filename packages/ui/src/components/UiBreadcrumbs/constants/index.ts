export const LINKS = [
  { url: '/', title: 'Main' },
  { url: '/category', title: 'Categories' },
  { url: '/cpu', title: 'CPU' },
];

export const DEFAULT_COLOR = 'default';
export const DEFAULT_ARIA_LABEL = 'Breadcrumb';
