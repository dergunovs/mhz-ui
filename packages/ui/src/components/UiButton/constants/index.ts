export const DEFAULT_LAYOUT = 'primary';
export const DEFAULT_TYPE = 'button';
export const DEFAULT_ICON = undefined;

export const DEFAULT_SLOT = 'Button label';
export const DEFAULT_ARIA_LABEL = 'Button';
