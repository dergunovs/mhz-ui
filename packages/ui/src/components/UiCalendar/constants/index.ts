import { ICalendarEvent } from '../interface';

export const EVENTS: ICalendarEvent<object>[] = [
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 1, text: 'Text' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 36%, rgb(235, 0, 123) 69%, rgb(251, 185, 0) 100%)',
  },
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 2, text: 'Text 2' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 41%, rgb(235, 0, 123) 71%, rgb(0, 134, 255) 100%)',
  },
];
