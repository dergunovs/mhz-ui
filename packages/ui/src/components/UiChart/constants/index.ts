export const LABELS = [
  '07.07 - 13.07',
  '14.07 - 20.07',
  '21.07 - 27.07',
  '28.07 - 03.08',
  '04.08 - 10.08',
  '11.08 - 17.08',
  '18.08 - 24.08',
];

export const DATASETS = [
  {
    data: [28, 16, 12, 4, 22, 16, 14],
    label: 'Грудь',
    borderColor: '#00CED1',
    backgroundColor: '#00CED1',
  },
  {
    data: [4, 6, 0, 2, 2, 0, 6],
    label: 'Ноги',
    borderColor: '#000080',
    backgroundColor: '#000080',
  },
  {
    data: [22, 18, 8, 10, 25, 18, 16],
    label: 'Плечи',
    borderColor: '#C41E3A',
    backgroundColor: '#C41E3A',
  },
  {
    data: [0, 6, 0, 6, 5, 4, 2],
    label: 'Пресс',
    borderColor: '#32CD32',
    backgroundColor: '#32CD32',
  },
  {
    data: [32, 24, 12, 10, 33, 24, 20],
    label: 'Руки',
    borderColor: '#DA70D6',
    backgroundColor: '#DA70D6',
  },
  {
    data: [16, 18, 8, 12, 26, 22, 20],
    label: 'Спина',
    borderColor: '#FF8C00',
    backgroundColor: '#FF8C00',
  },
];

export const TYPE = 'Line';
export const TITLE = 'Chart title';
