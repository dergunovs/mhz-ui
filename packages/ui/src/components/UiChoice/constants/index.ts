export const options = [1, 2, 3, 4, 5];
export const modelValue = options[2];
export const title = 'Заголовок';
export const isTall = true;
export const isInput = true;
