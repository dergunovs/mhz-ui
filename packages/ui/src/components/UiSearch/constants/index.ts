export const DEBOUNCE_TIME = 300;

export const MODEL_VALUE = '';
export const LANG = 'ru';
export const IS_SUCCESS = true;

export const RESULTS = {
  products: [
    { _id: '1', title: 'AMD Ryzen 7700X' },
    { _id: '2', title: 'AMD Ryzen 7900X' },
  ],
  manufacturers: [{ _id: '1', title: 'AMD' }],
};

export const SEARCH_SCHEME = [
  { type: 'products', labels: ['title'], url: '/products' },
  { type: 'manufacturers', labels: ['title'], url: '/manufacturers' },
];
