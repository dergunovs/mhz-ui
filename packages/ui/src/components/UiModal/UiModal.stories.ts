import { Meta, StoryObj } from '@storybook/vue3';
import { html } from 'mhz-helpers';

import { MODEL_VALUE, IS_CONFIRM } from './constants';

import { UiModal, UiButton } from '@/components';

const meta: Meta<typeof UiModal> = {
  component: UiModal,
  args: {
    modelValue: MODEL_VALUE,
    isConfirm: IS_CONFIRM,
  },
  argTypes: {
    lang: {
      options: ['en', 'ru'],
    },
  },
};

export default meta;

export const Primary: StoryObj<typeof UiModal> = {
  render: (args, { updateArgs }) => ({
    components: { UiModal, UiButton },
    setup: () => ({ args }),

    template: html`<div>
      <UiButton @click="update">Show modal</UiButton>
      <UiModal v-bind="args" @update:modelValue="update">Text inside</UiModal>
    </div>`,

    methods: {
      update(value: boolean) {
        updateArgs({ modelValue: value });
      },
    },
  }),
};
