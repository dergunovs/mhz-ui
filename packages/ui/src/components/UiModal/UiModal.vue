<template>
  <Teleport to="body">
    <div v-if="props.modelValue" :class="$style.backdrop" @mousedown="hide" data-test="ui-modal-backdrop">
      <dialog ref="dialogRef" :class="$style.dialog" @cancel.prevent="hide" v-bind="$attrs" data-test="ui-modal-dialog">
        <div :class="$style.modal" @mousedown.stop :data-scrollable="props.isScrollable" data-test="ui-modal">
          <div :class="$style.header">
            <UiClose @click="hide" data-test="ui-modal-close" />
          </div>

          <div :class="$style.slot" data-test="ui-modal-slot">
            <slot></slot>
          </div>

          <div v-if="props.isConfirm" :class="$style.buttons">
            <UiButton @click="handleConfirm" data-test="ui-modal-confirm">{{ MESSAGES[props.lang].confirm }}</UiButton>

            <UiButton layout="secondary" @click="hide" data-test="ui-modal-cancel">
              {{ MESSAGES[props.lang].cancel }}
            </UiButton>
          </div>
        </div>
      </dialog>
    </div>
  </Teleport>
</template>

<script setup lang="ts">
import { ref, watch, computed, nextTick, onMounted } from 'vue';

import UiButton from '../UiButton/UiButton.vue';
import UiClose from '../UiClose/UiClose.vue';

import { TLocale } from '@/components/locales/types';
import { MESSAGES } from '@/components/locales';

interface IProps {
  modelValue: boolean;
  isScrollable?: boolean;
  isConfirm?: boolean;
  width?: string;
  lang?: TLocale;
}

interface IEmit {
  'update:modelValue': [value: boolean];
  confirm: [];
}

const props = withDefaults(defineProps<IProps>(), {
  width: '360',
  lang: 'ru',
});

const emit = defineEmits<IEmit>();

defineOptions({ inheritAttrs: false });

const widthComputed = computed(() => `${props.width}px`);
const dialogRef = ref<HTMLDialogElement | null>(null);

function toggleDialog() {
  nextTick(() => {
    if (dialogRef.value?.open) {
      dialogRef.value.close();
    } else {
      dialogRef.value?.showModal();
    }
  });
}

function handleConfirm() {
  emit('confirm');
  hide();
}

function hide() {
  emit('update:modelValue', false);
}

onMounted(() => {
  watch(
    () => props.modelValue,
    () => {
      toggleDialog();
    },
    { immediate: true }
  );
});
</script>

<style module lang="scss">
.backdrop {
  position: fixed;
  inset: 0;
  z-index: 998;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: var(--color-gray-transparent);
}

.dialog {
  z-index: 999;
  padding: 0;
  outline: none;
  background-color: var(--color-white);
  border: none;
  border-radius: 16px;
}

.modal {
  display: flex;
  flex-direction: column;
  gap: 12px;
  width: v-bind(widthComputed);
  max-width: 100%;
  height: auto;
  padding: 24px;
  box-shadow: 0 0 16px 0 var(--color-gray-transparent);

  &[data-scrollable='true'] {
    max-height: 64dvh;
    overflow-y: auto;
  }
}

.header {
  display: flex;
  justify-content: flex-end;
}

.buttons {
  display: flex;
  gap: 16px;
}

:global(.dark) {
  .modal {
    background-color: var(--color-black);
    box-shadow: 0 0 16px 0 var(--color-black-transparent);
  }

  .slot {
    color: var(--color-white);
  }
}
</style>
