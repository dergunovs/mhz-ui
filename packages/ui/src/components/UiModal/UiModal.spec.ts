import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest, wait } from 'mhz-helpers';

import UiModal from './UiModal.vue';
import { IS_CONFIRM, DEFAULT_SLOT } from './constants';

import { wrapperFactory } from '@/test';

const modal = dataTest('ui-modal');
const modalDialog = dataTest('ui-modal-dialog');
const modalBackdrop = dataTest('ui-modal-backdrop');
const modalClose = dataTest('ui-modal-close');
const modalSlot = dataTest('ui-modal-slot');
const modalConfirm = dataTest('ui-modal-confirm');
const modalCancel = dataTest('ui-modal-cancel');

let wrapper: VueWrapper<InstanceType<typeof UiModal>>;

beforeEach(() => {
  wrapper = wrapperFactory(UiModal, { modelValue: false, isConfirm: IS_CONFIRM }, { default: DEFAULT_SLOT });
});

enableAutoUnmount(afterEach);

describe('UiModal', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(UiModal)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows modal by modelValue props', async () => {
    expect(wrapper.find(modalBackdrop).exists()).toBe(false);
    expect(wrapper.find(modalDialog).exists()).toBe(false);

    await wrapper.setProps({ modelValue: true });
    await wait(100);

    expect(wrapper.find(modalBackdrop).exists()).toBe(true);
    expect(wrapper.find(modalDialog).exists()).toBe(true);
  });

  it('hides modal by close button click', async () => {
    await wrapper.setProps({ modelValue: true });
    await wait(100);

    expect(wrapper.find(modalClose).exists()).toBe(true);

    await wrapper.find(modalClose).trigger('click');

    expect(wrapper.emitted('update:modelValue')).toHaveLength(1);
    expect(wrapper.emitted('update:modelValue')?.[0]).toEqual([false]);
  });

  it('hides modal by backdrop click', async () => {
    await wrapper.setProps({ modelValue: true });
    await wait(100);

    expect(wrapper.find(modalBackdrop).exists()).toBe(true);

    await wrapper.find(modalBackdrop).trigger('mousedown');

    expect(wrapper.emitted('update:modelValue')).toHaveLength(1);
    expect(wrapper.emitted('update:modelValue')?.[0]).toEqual([false]);
  });

  it('shows slot content', async () => {
    await wrapper.setProps({ modelValue: true });
    await wait(100);

    expect(wrapper.find(modalSlot).text()).toBe(DEFAULT_SLOT);
  });

  it('shows confirm and cancel buttons in confirm mode', async () => {
    await wrapper.setProps({ modelValue: true, isConfirm: true });
    await wait(100);

    expect(wrapper.find(modalConfirm).exists()).toBe(true);
    expect(wrapper.find(modalCancel).exists()).toBe(true);
  });

  it('sets lang for buttons', async () => {
    await wrapper.setProps({ modelValue: true, isConfirm: true });
    await wait(100);

    expect(wrapper.findComponent(modalConfirm).text()).toBe('Подтвердить');
    expect(wrapper.findComponent(modalCancel).text()).toBe('Отмена');

    await wrapper.setProps({ lang: 'en' });
    await wait(100);

    expect(wrapper.findComponent(modalConfirm).text()).toBe('Confirm');
    expect(wrapper.findComponent(modalCancel).text()).toBe('Cancel');
  });

  it('hides by cancel button click in confirm mode', async () => {
    await wrapper.setProps({ modelValue: true, isConfirm: true });
    await wait(100);

    await wrapper.find(modalCancel).trigger('click');

    expect(wrapper.emitted()).not.toHaveProperty('confirm');
    expect(wrapper.emitted('update:modelValue')).toHaveLength(1);
    expect(wrapper.emitted('update:modelValue')?.[0]).toEqual([false]);
  });

  it('emit confirm and hides by confirm button click in confirm mode', async () => {
    await wrapper.setProps({ modelValue: true, isConfirm: true });
    await wait(100);

    await wrapper.find(modalConfirm).trigger('click');

    expect(wrapper.emitted('confirm')).toHaveLength(1);
    expect(wrapper.emitted('update:modelValue')).toHaveLength(1);
    expect(wrapper.emitted('update:modelValue')?.[0]).toEqual([false]);
  });

  it('closes on escape key press (cancel event)', async () => {
    await wrapper.setProps({ modelValue: true });
    await wait(100);

    const dialogElement = wrapper.find(modalDialog).element as HTMLDialogElement;

    expect(dialogElement.open).toBe(true);

    dialogElement.dispatchEvent(new Event('cancel', { cancelable: true }));

    expect(wrapper.emitted('update:modelValue')).toHaveLength(1);
    expect(wrapper.emitted('update:modelValue')?.[0]).toEqual([false]);
  });

  it('toggles dialog correctly on modelValue change', async () => {
    await wrapper.setProps({ modelValue: true });
    await wait(100);

    const dialogElement = wrapper.find(modalDialog).element as HTMLDialogElement;

    expect(dialogElement.open).toBe(true);

    await wrapper.setProps({ modelValue: false });
    await wait(100);

    const updateEvents = wrapper.emitted('update:modelValue');

    if (updateEvents) expect(updateEvents.at(-1)).toEqual([false]);
  });

  it('renders with correct scrollable attribute', async () => {
    await wrapper.setProps({ modelValue: true, isScrollable: true });
    await wait(100);

    expect(wrapper.find(modal).attributes('data-scrollable')).toBe('true');

    await wrapper.setProps({ isScrollable: false });
    await wait(100);

    expect(wrapper.find(modal).attributes('data-scrollable')).toBe('false');
  });
});
