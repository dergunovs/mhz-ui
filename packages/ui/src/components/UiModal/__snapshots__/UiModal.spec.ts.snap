// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UiModal > matches snapshot 1`] = `"<!--v-if-->"`;
